src/main/jhipster/src/main/webapp/app/shared/tree/study/study-tree.component.ts

```diff
@@ -14,7 +16,10 @@ export class StudyTreeComponent extends TreeComponent {
     title = 'Browse for studies';
 
     constructor(public service: TreeService,
-                public activeModal: NgbActiveModal) {
-        super(service, activeModal);
+                public activeModal: NgbActiveModal,
+                public alertService: AlertService,
+                public translateService: TranslateService,
+                public modalService: NgbModal) {
+        super(false, service, activeModal, alertService, translateService, modalService);
```

Comment:
It won't affect because because studytree component uses tree.component.html which still don't have the folder management controls, but perhaps it can be constructed readonly=true since StudyTreeService methods are not implemented yet.
But as you mentioned to me earlier, this is a shortcoming of the _constructor_ approach, maybe we can evaluate later if it would be possible to remove it from the constructor params and set it as a normal variable while opening the modal. This will open the possibility to use the same component (tree or tree table) both in readonly/not-readonly mode